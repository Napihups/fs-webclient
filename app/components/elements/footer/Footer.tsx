import { FIXTURES } from "@constant/fixtures";
import { LogoPlainWhite } from "@element/Logo/LogoPlainWhite";
import React from "react";

export const Footer: React.FC = () => {
  return (
    <footer
      data-testid="app-footer"
      aria-label="Famspace Company metadata"
      className="footer footer-center bg-primary text-primary-content p-10 border-t-2 border-t-base-200"
    >
      <div>
        <LogoPlainWhite />
        <p className="font-bold">{FIXTURES.footer.company_name}</p>
        <p>{FIXTURES.footer.copyright}</p>
      </div>
    </footer>
  );
};
