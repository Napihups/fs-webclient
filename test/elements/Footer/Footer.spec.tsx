import { render, screen } from "@testing-library/react";
import { Footer } from "@element/Footer/Footer";

describe("Footer Static UI Test &  Snapshot", () => {
  test("Static: textual test on Footer", async () => {
    render(<Footer />);

    await screen.getByText("Napigo Pte Ltd.");
    await screen.getByText("Copyright Â© 2022 - All right reserved");
  });
  test("Static: Footer background should be using primary color", async () => {
    render(<Footer />);

    const footer = await await screen.findByTestId("app-footer");
    expect(footer).toHaveClass("bg-primary");
  });
  test("Static: Footer Snapshot changes test", async () => {
    await render(<Footer />);

    const footer = await await screen.findByTestId("app-footer");
    expect(footer).toMatchSnapshot();
  });
});
