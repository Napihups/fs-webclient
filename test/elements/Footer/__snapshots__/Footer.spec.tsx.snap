// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Static UI Test &  Snapshot Static: Footer Snapshot changes test 1`] = `
<footer
  aria-label="Famspace Company metadata"
  class="footer footer-center bg-primary text-primary-content p-10 border-t-2 border-t-base-200"
  data-testid="app-footer"
>
  <div>
    <div
      class="scale-75"
    >
      <svg
        fill="none"
        height="60"
        width="229"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.995 9.073a.968.968 0 0 0-1.228-.654c-4.969 1.659-9.23 4.918-12.076 9.26A22.425 22.425 0 0 0 .238 33.252c.799 5.43 3.569 10.394 7.801 13.984 4.233 3.59 9.646 5.566 15.245 5.564 5.6-.002 11.01-1.982 15.24-5.575 4.23-3.594 6.996-8.561 7.79-13.99a22.426 22.426 0 0 0-3.465-15.57c-2.85-4.34-7.114-7.597-12.084-9.252a.968.968 0 0 0-1.228.655l-2.57 8.59c-.16.533.15 1.09.67 1.287a12.051 12.051 0 0 1 5.813 4.642 11.658 11.658 0 0 1 1.802 8.095 11.788 11.788 0 0 1-4.05 7.272 12.248 12.248 0 0 1-7.923 2.898 12.249 12.249 0 0 1-7.924-2.892 11.788 11.788 0 0 1-4.056-7.27 11.658 11.658 0 0 1 1.795-8.095 12.05 12.05 0 0 1 5.81-4.647c.52-.197.828-.755.668-1.288l-2.577-8.587ZM75.18 21.205c3.03 0 5.46.9 7.29 2.7 1.86 1.8 2.79 4.515 2.79 8.145V46h-8.55V33.445c0-3.33-1.335-4.995-4.005-4.995-1.47 0-2.655.48-3.555 1.44-.87.96-1.305 2.4-1.305 4.32V46h-8.55V21.61h8.145v2.655a9.44 9.44 0 0 1 3.42-2.25c1.32-.54 2.76-.81 4.32-.81Zm41.584.405V46h-8.145v-2.43c-1.59 1.89-3.99 2.835-7.2 2.835-2.22 0-4.245-.51-6.075-1.53-1.8-1.05-3.24-2.535-4.32-4.455-1.05-1.92-1.575-4.14-1.575-6.66 0-2.52.525-4.725 1.575-6.615 1.08-1.92 2.52-3.39 4.32-4.41 1.83-1.02 3.855-1.53 6.075-1.53 2.88 0 5.145.84 6.795 2.52V21.61h8.55Zm-13.545 18c1.5 0 2.73-.51 3.69-1.53.96-1.05 1.44-2.49 1.44-4.32 0-1.8-.48-3.21-1.44-4.23-.96-1.02-2.19-1.53-3.69-1.53s-2.73.51-3.69 1.53-1.44 2.43-1.44 4.23c0 1.83.48 3.27 1.44 4.32.96 1.02 2.19 1.53 3.69 1.53Zm34.394-18.405c2.22 0 4.23.525 6.03 1.575 1.83 1.02 3.27 2.49 4.32 4.41 1.08 1.89 1.62 4.095 1.62 6.615 0 2.52-.54 4.74-1.62 6.66-1.05 1.89-2.49 3.36-4.32 4.41-1.8 1.02-3.81 1.53-6.03 1.53-2.85 0-5.115-.855-6.795-2.565v10.89h-8.55V21.61h8.145v2.43c1.59-1.89 3.99-2.835 7.2-2.835Zm-1.8 18.405c1.5 0 2.73-.51 3.69-1.53.96-1.05 1.44-2.475 1.44-4.275 0-1.8-.48-3.21-1.44-4.23-.96-1.05-2.19-1.575-3.69-1.575s-2.73.525-3.69 1.575c-.96 1.02-1.44 2.43-1.44 4.23 0 1.8.48 3.225 1.44 4.275.96 1.02 2.19 1.53 3.69 1.53Zm17.744-18h8.55V46h-8.55V21.61Zm4.275-2.7c-1.56 0-2.82-.435-3.78-1.305-.96-.87-1.44-1.95-1.44-3.24 0-1.29.48-2.37 1.44-3.24.96-.87 2.22-1.305 3.78-1.305 1.56 0 2.82.42 3.78 1.26.96.81 1.44 1.86 1.44 3.15 0 1.35-.48 2.475-1.44 3.375-.96.87-2.22 1.305-3.78 1.305Zm36.067 2.7v19.935c0 4.53-1.23 7.92-3.69 10.17-2.46 2.28-6 3.42-10.62 3.42-2.4 0-4.65-.27-6.75-.81-2.1-.54-3.885-1.335-5.355-2.385l3.105-5.985c.99.81 2.235 1.44 3.735 1.89 1.53.48 3.03.72 4.5.72 2.25 0 3.9-.495 4.95-1.485 1.05-.99 1.575-2.46 1.575-4.41v-.81c-1.65 1.92-4.065 2.88-7.245 2.88-2.16 0-4.155-.48-5.985-1.44a11.352 11.352 0 0 1-4.365-4.14c-1.08-1.8-1.62-3.87-1.62-6.21 0-2.31.54-4.35 1.62-6.12 1.08-1.8 2.535-3.18 4.365-4.14 1.83-.99 3.825-1.485 5.985-1.485 3.48 0 6.03 1.125 7.65 3.375v-2.97h8.145Zm-13.77 16.335c1.53 0 2.79-.465 3.78-1.395 1.02-.93 1.53-2.13 1.53-3.6 0-1.47-.495-2.655-1.485-3.555-.99-.93-2.265-1.395-3.825-1.395s-2.85.465-3.87 1.395c-.99.9-1.485 2.085-1.485 3.555 0 1.47.51 2.67 1.53 3.6 1.02.93 2.295 1.395 3.825 1.395Zm31.503 8.46c-2.61 0-4.965-.54-7.065-1.62-2.1-1.08-3.75-2.58-4.95-4.5-1.17-1.92-1.755-4.095-1.755-6.525 0-2.4.585-4.56 1.755-6.48 1.2-1.92 2.835-3.405 4.905-4.455 2.1-1.08 4.47-1.62 7.11-1.62 2.64 0 5.01.54 7.11 1.62 2.1 1.05 3.735 2.535 4.905 4.455 1.17 1.89 1.755 4.05 1.755 6.48 0 2.43-.585 4.605-1.755 6.525s-2.805 3.42-4.905 4.5-4.47 1.62-7.11 1.62Zm0-6.795c1.5 0 2.73-.51 3.69-1.53.96-1.05 1.44-2.49 1.44-4.32 0-1.8-.48-3.21-1.44-4.23-.96-1.02-2.19-1.53-3.69-1.53s-2.73.51-3.69 1.53-1.44 2.43-1.44 4.23c0 1.83.48 3.27 1.44 4.32.96 1.02 2.19 1.53 3.69 1.53Z"
          fill="#fff"
        />
      </svg>
    </div>
    <p
      class="font-bold"
    >
      Napigo Pte Ltd.
    </p>
    <p>
      Copyright Â© 2022 - All right reserved
    </p>
  </div>
</footer>
`;
